src/test/java/org/elasticsearch/indices/IndicesServiceTest.java

```diff
@@ -143,14 +138,23 @@ public void testPendingTasks() throws IOException {
         for (Path p : paths) {
             assertTrue(Files.exists(p));
         }
-        indicesService.addPendingDelete(test.index(), new ShardId(test.index(), 0), test.getIndexSettings());
+        int numPending = 1;
+        if (randomBoolean()) {
+            indicesService.addPendingDelete(test.index(), new ShardId(test.index(), 0), test.getIndexSettings());
+        } else {
+            if (randomBoolean()) {
+                numPending++;
+                indicesService.addPendingDelete(test.index(), new ShardId(test.index(), 0), test.getIndexSettings());
+            }
+            indicesService.addPendingDelete(test.index(), test.getIndexSettings());
```

Comment:
I know it's a pain code wise, but can we sometime put the index delete first in the queue?